pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -10,18 +10,22 @@ import (
 // KeycloakSpec defines the desired state of Keycloak
 // +k8s:openapi-gen=true
 type KeycloakSpec struct {
-	ExternalDatabaseSecret string                 `json:"externalDatabaseSecret,omitempty"`
-	AdminCredentialSecret  string                 `json:"adminCredentialSecret,omitempty"`
-	Extensions             []string               `json:"extensions,omitempty"`
-	Instances              int                    `json:"instances,omitempty"`
-	ExternalAccess         KeycloakExternalAccess `json:"externalAccess,omitempty"`
-	Profile                string                 `json:"profile,omitempty"`
+	AdminCredentialSecret string                   `json:"adminCredentialSecret,omitempty"`
+	Extensions            []string                 `json:"extensions,omitempty"`
+	Instances             int                      `json:"instances,omitempty"`
+	ExternalAccess        KeycloakExternalAccess   `json:"externalAccess,omitempty"`
+	ExternalDatabase      KeycloakExternalDatabase `json:"externalDatabase,omitempty"`
+	Profile               string                   `json:"profile,omitempty"`
 }
 
 type KeycloakExternalAccess struct {
 	Enabled bool `json:"enabled,omitempty"`
 }
 
+type KeycloakExternalDatabase struct {
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
It would be good if we allowed the end user to supply a secret name here instead of the boolean. my concern is that the secret name needs to match what we currently call the secret in the operator. If this is left as is, it needs to be pointed out in the docs that the secret needs a specific name.

Reply:
I've been talking about this on our internal channel. The conclusion was the we can either reference a secret or reuse the same we would create in hosted database mode. 

I decided to for the latter as the integration code is already there.

I also tried to be very explicit about the expected secret name in the docs. Could you please have a look at them and tell me, what parts could be improved?

Reply:
I agree, reusing the same secret is perfect and should definitely be done. Allowing them to specify a different name to this existing secret is more of an enhancement and not required for now. It is very clear in the docs, thank you

Reply:
I think there would still be benefit in having a configurable secret name. What about reading `DatabaseSecretName` from the CR (see e.g. https://github.com/keycloak/keycloak-operator/blob/master/pkg/model/postgresql_deployment.go#L77) and set it to a default value of `<profile>-db-secret`?

Reply:
I'm still not convinced to be honest. We allow only one Keycloak installation per namespace, so there can be only one database secret. I can't what could making the secret name configurable buy us. 

So I would propose to do the following - if you have in mind a usecase, where we need it - please let me know. I'll add it then. If you think more of a future and some possible usecases - let's just create a JIRA for it. This way we could implement it later. Would that work for you?

Reply:
+1 from me